rclcpp/test/rclcpp/test_publisher.cpp

```diff
@@ -296,6 +296,20 @@ TEST_F(TestPublisher, basic_getters) {
   }
 }
 
+TEST_F(TestPublisher, intra_process_comm_state_publisher) {
```

Comment:
what about adding the test with `IntraProcessSetting::NodeDefault`? (this goes to subscription as well.)

Reply:
Sure it's possible to add a test, according the [default node options](https://github.com/ros2/rclcpp/blob/rolling/rclcpp/include/rclcpp/node_options.hpp#L46) `rclcpp::NodeOptions` would return `false` for the intra-process state. The [init method](https://github.com/ros2/rclcpp/blob/rolling/rclcpp/test/rclcpp/test_publisher.cpp#L47-L50) for the `node` is just sets up a basic, default `NodeOptions`. So I could add the one test, for the way it's setup now, and just expect false. Then, if wanted I could create another `initialize` method which would change the default. Personally I think the latter is unnecessary, but the former is still valid.  Thoughts?

Reply:
yes, i think so. so that we can detect the test or default behavior change for this.